Tools/DeployApps/BaasClient.cs

```diff
@@ -863,6 +881,154 @@ private static class Schemas
+
+                    if (existingContainer is not null)
+                    {
+                        output.WriteLine($"Container with id {existingContainer.ContainerId} found.");
+
+                        if (!existingContainer.IsRunning)
+                        {
+                            output.WriteLine($"Waiting for container with id {existingContainer.ContainerId} to be running.");
+                            await WaitForContainer(existingContainer.ContainerId);
+                        }
+
+                        return existingContainer.HttpUrl;
+                    }
+                }
+
+                output.WriteLine($"No container found, starting a new one.");
+                var containerId = await StartContainer(differentiator);
+
+                output.WriteLine($"Container with id {containerId} started, waiting for it to be running.");
+                var container = await WaitForContainer(containerId);
+
+                return container.HttpUrl;
+            }
+
+            private Task<ContainerInfo[]?> GetContainers()
+            {
+                return CallEndpointAsync<ContainerInfo[]>(HttpMethod.Get, "listContainers");
+            }
+
+            private Task StopContainer(string id)
+            {
+                return CallEndpointAsync<BsonDocument>(HttpMethod.Post, $"stopContainer?id={id}");
+            }
+
+            private async Task<string?> GetCurrentUserId()
+            {
+                return (await CallEndpointAsync<BsonDocument>(HttpMethod.Get, "userinfo"))!["id"].AsString;
+            }
+
+            private async Task<string> StartContainer(string differentiator)
+            {
+                var response = await CallEndpointAsync<BsonDocument>(HttpMethod.Post, "startContainer", new[]
+                {
+                    new
+                    {
+                        key = "DIFFERENTIATOR",
+                        value = differentiator,
+                    }
+                });
+
+                return response?["id"].AsString!;
+            }
+
+            private async Task<ContainerInfo> WaitForContainer(string containerId, int maxRetries = 100)
+            {
+                while (maxRetries > 0)
+                {
+                    maxRetries -= 1;
+
+                    var containers = await GetContainers();
+                    var container = containers!.FirstOrDefault(c => c.ContainerId == containerId);
+
+                    if (container?.IsRunning == true)
+                    {
+                        return container;
+                    }
+
+                    await Task.Delay(3000);
+                }
+
+                throw new Exception($"Container with id={containerId} was not found or ready after {maxRetries} retrues");
+            }
+
+            private async Task<T?> CallEndpointAsync<T>(HttpMethod method, string relativePath, object? payload = null)
```

Comment:
This is almost the same as `BaasClient.SendAsync`, without the refresh token capabilities. We could remove this